ios/RNFirebase/links/RNFirebaseLinks.m

```diff
@@ -2,47 +2,174 @@
 #import "Firebase.h"
 
 #if __has_include(<FirebaseDynamicLinks/FIRDynamicLink.h>)
+#import "RNFirebaseEvents.h"
+
+
+static void sendDynamicLink(NSURL *url, id sender) {
+    [[NSNotificationCenter defaultCenter] postNotificationName:LINKS_DYNAMIC_LINK_RECEIVED
+                                                        object:sender
+                                                      userInfo:@{@"url": url.absoluteString}];
+    NSLog(@"sendDynamicLink Success: %@", url.absoluteString);
+}
+
 @implementation RNFirebaseLinks
 
 RCT_EXPORT_MODULE();
 
+- (id)init {
+    self = [super init];
+    if (self != nil) {
+        NSLog(@"Setting up RNFirebaseLinks instance");
+        [self initialiseLinks];
+    }
+    return self;
+}
+
+- (void)initialiseLinks {
+    // Set up internal listener to send notification over bridge
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(sendDynamicLinkEvent:)
+                                                 name:LINKS_DYNAMIC_LINK_RECEIVED
+                                               object:nil];
+}
+
+- (void)dealloc {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
++ (BOOL)application:(UIApplication *)app
+            openURL:(NSURL *)url
+            options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options {
+    FIRDynamicLink *dynamicLink =
+    [[FIRDynamicLinks dynamicLinks] dynamicLinkFromCustomSchemeURL:url];
```

Comment:
I think you can do `return [self application:app openURL:url sourceApplication:nil annotation:@{}];` or something similar here